### PR TITLE
PVP 290 Changed toasts to snackbars

### PR DESCRIPTION
**Pull request by vytjar**
- Modified Composables that used toasts to be made of scaffolds that allow showing of snackbars
- Created CenteredSnackbarHost that centers the text which looks nicer

This is how they look now:

image

### CODE REVIEW

**Comment by klaidoshka** [comment]
Cannot test it myself, so I will just give some comments and for now accept it, as I believe it works to some extent.

From user perspective.
- How is it being closed? By swiping left/right/up/down? Or some other specific way?
- Could we add a close button to the very row end for aesthetics and intuitive close function?
  - And leave left/right swipe to close it too, if it already exists. If not, would be nice to have for experienced users.
- Maybe our logo would be nice to the very row start?

From code perspective.
- Couldn't we just add scaffold to the LayoutBootstrap and propagate `showScaffold` function to the children components via `CompositionLocalProvider`? This would save us from nesting each screen into additional scaffold container, which is only used for snackbar, where one singular snackbar is sufficient (I believe) and there is no need to have different ones for each screen.

**Comment by vytjar** [comment]
> * How is it being closed? By swiping left/right/up/down? Or some other specific way?
> * Could we add a close button to the very row end for aesthetics and intuitive close function?

It closes itself, just like a toast.

**Comment by vytjar** [comment]
> * Couldn't we just add scaffold to the LayoutBootstrap and propagate `showScaffold` function to the children components via `CompositionLocalProvider`? This would save us from nesting each screen into additional scaffold container, which is only used for snackbar, where one singular snackbar is sufficient (I believe) and there is no need to have different ones for each screen.

Thanks, didn't know about that. Seems to work.

File: app/src/main/java/com/pvp/app/ui/screen/drawer/DrawerScreen.kt [hunk 1]
```diff
@@ -149,13 +149,16 @@ private fun Header(modifier: Modifier = Modifier) {
     }
 }
 
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/drawer/DrawerScreen.kt:152]: +1

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 2]
```diff
@@ -72,13 +71,14 @@ private val INGREDIENTS = Ingredient.entries.map { it.title }
 @Composable
 private fun AccountDeleteButton(
     viewModel: ProfileViewModel = hiltViewModel(),
-    state: ProfileState
+    state: ProfileState,
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:74]: +1

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 3]
```diff
@@ -268,7 +269,9 @@ fun ProfileScreen(
     ) {
         Points(points = state.user.points)
 
-        Column(modifier = Modifier.fillMaxWidth()) {
+        Column(
+            modifier = Modifier.fillMaxWidth()
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:273]: +1 (unless doesn't fit)

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 4]
```diff
@@ -255,6 +255,7 @@ fun ProfileScreen(
 ) {
     val state by viewModel.state.collectAsStateWithLifecycle()
 
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:258]: +1

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 5]
```diff
@@ -291,7 +294,7 @@ fun ProfileScreen(
                 onUpdateIngredients = { viewModel.update { u -> u.ingredients = it } },
                 onUpdateHeight = { viewModel.update { u -> u.height = it } },
                 onUpdateMass = { viewModel.update { u -> u.mass = it } },
-                state = state
+                state = state,
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:297]: +1

File: app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt [hunk 6]
```diff
@@ -306,15 +309,14 @@ private fun Properties(
     onUpdateHeight: (Int) -> Unit,
     onUpdateIngredients: (List<Ingredient>) -> Unit,
     onUpdateMass: (Int) -> Unit,
-    state: ProfileState
+    state: ProfileState,
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt:312]: +1

File: app/src/main/java/com/pvp/app/ui/screen/settings/SettingsScreen.kt [hunk 7]
```diff
@@ -255,13 +255,16 @@ private fun SettingHealthConnectPermissions(context: Context) {
 }
 
 @Composable
-private fun GoogleCalendarSynchronizer(model: SettingsViewModel = hiltViewModel()) {
+private fun GoogleCalendarSynchronizer(
+    model: SettingsViewModel = hiltViewModel(),
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/settings/SettingsScreen.kt:259]: +1

**Review by AurisTFG — APPROVED** [review]

**Pull request merged**